### ISSUE TITLE
Get context dynamically inside a dynamic partial call

### ISSUE THREAD

**Issue by landwire** [wycats/handlebars.js#1494]
Before filing issues, please check the following points first:

- [x] Have a look at https://github.com/wycats/handlebars.js/blob/master/CONTRIBUTING.md
- [x] Read the FAQ at https://github.com/wycats/handlebars.js/blob/master/FAQ.md
- [ ] Use the jsfiddle-template at https://jsfiddle.net/9D88g/47/ to reproduce problems or bugs

Sorry, I cannot create this using the fiddle.

The yml data looks like this:
```
items:
  - component: f-radio
    data: f-radio--flex
```
I am using handlebars with assemble and render some partials dynamically. This works great. {{ data }} also prints out f-radio--flex, so I have the correct value in data 
```
{{#each items}}
  {{data}}
  {{> (lookup . 'component') }}
{{/each}}
```
 (https://handlebarsjs.com/partials.html)

Now I am trying to add partial context/data dynamically as well. All data sits at the root context.
So I tried it with the fixed context first: `{{> (lookup . 'component') @root.f-radio--flex }}` and this works as expected and uses the correct data and renders the partial.

Then dynamically I tried:
`{{> (lookup . 'component') (lookup . 'data') }}` => renders but no data is passed into the partial
`{{> (lookup . 'component') (lookup @root 'data') }}` => renders but no data is passed into the partial
`{{> (lookup . 'component') @root.(lookup @root 'data') }}` => throws an error
```
Warning: Parse error on line 5:
... 'component') @root.(lookup . 'data') }}
-----------------------^
Expecting 'ID', got 'OPEN_SEXPR' Use --force to continue.
```

None of which rendered the correct result.
So basically the question is how can I express this line dynamically based on the data from the yml file above: `{{> (lookup . 'component') @root.f-radio--flex }}`

**Comment by doowb** [comment]
@landwire have you tried either of these:

```handlebars
{{> (lookup . 'component') data }}
{{> (lookup . 'component') this.data }}
```

Those should work since `data` is in the current scope of the `each` block and it's not something that needs to be dynamically looked up.

**Comment by landwire** [comment]
> @landwire have you tried either of these:
> 
> ```
> {{> (lookup . 'component') data }}
> {{> (lookup . 'component') this.data }}
> ```
> 
> Those should work since `data` is in the current scope of the `each` block and it's not something that needs to be dynamically looked up.

Thanks. But the data needs to be read out of the yml file called f-radio--flex. Maybe that was not clear. So rather than writing:
```
data:
  label: Anrede
  type: radio
  description: Bitte wählen Sie Ihre Anrede aus
.....
```
I just want to reference a file where all this data is inside.
Here is the content of the yml file:

```
---
label: Anrede
type: radio
description: Bitte wählen Sie Ihre Anrede aus

inputgroup:
  id: input-radios-group-1

  attributes:
    required: true
    type: radio
    name: input-radios-group-1

  inputs:
    - label: Label 1
      attributes:
        id: field-group-1-input-radio-1
        value: Value 1
        checked: checked

    - label: Label 2
      attributes:
        id: field-group-1-input-radio-2
        value: Value 2

    - label: Label 3
      attributes:
        id: field-group-1-input-radio-3
        value: Value 3

    - label: Label 4
      attributes:
        id: field-group-1-input-radio-4
        value: Value 4

    - label: Label 5
      attributes:
        id: field-group-1-input-radio-5
        value: Value 5

    - label: Label 6
      attributes:
        id: field-group-1-input-radio-6
        value: Value 6
```

Sorry, I was unclear before.

**Comment by doowb** [comment]
> Sorry, I was unclear before.

I misunderstood the issue... I missed the part where you said it worked when using `@root.f-radio--flex`.

I think this should work then:

```handlebars
{{> (lookup . 'component') (lookup @root data) }}
```

This is because the `lookup` helper wants the actual property name to "lookup" on the object `@root`. Since the property should be the name of the file specified by the value of `data`, then you should pass the value of `data` in. Using the quotes cause it to pass the actual literal "data" which was then trying to return `@root.data`.

**Comment by landwire** [comment]
Ok, thanks. Will try this tomorrow. Just assumed as ‘component’ works with quotes same would go for data.

**Comment by landwire** [comment]
@doowb Hey Brian, that worked! Thank you so so much! 

Is there a concise documentation about the lookup helper somewhere? I just do not understand why (lookup . 'component') finds a .hbs file and then (lookup . 'data') would not find the yml file. I also don't understand what the dot means exactly. 

In the documentation I could only find this example, which again has totally different syntax and hardly any explanation.

```
{{#each bar}}
  {{lookup ../foo @index}}
{{/each}}
```

Anyway, I am very grateful for your help!

**Comment by nknapp** [comment]
Closing due to inactivity

**Comment by bcairns** [comment]
Is there a way to do this when the data you want is a specific property in the file, versus the entire file?

Eg, we have a bunch of banners defined in a banners.json file, and this works: {{> banner banners.variant1 }}, but I can't figure out how to supply "banners.variant1" dynamically.

So I'd like to do something like {{> banner (lookup @root banner-data) }}, where banner-data = 'banners.variant1'.  But it doesn't work.

It only works when you simply specify the file, so if I had a single banner defined in banner-variant1.json, and banner-data = 'banner-variant1', the above code works.

Is there a way to accomplish what I'm trying to do?  Or is it better to avoid defining multiple datasets in a single JSON file?

**Comment by bcairns** [comment]
I have figured out a couple ways of doing this.

With JSON data like: { data-file: "banners", data-variant: "variant1" }, you do either:

```
{{#with (lookup @root data-file)}}
	{{> banner (lookup this ../data-variant) }}<br>
{{/with}}
```

or as one line with a nested lookup:

`{{> banner (lookup (lookup @root data-file) data-selector ) }}`
